src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -1216,14 +1215,15 @@ private void BlittableArrayToListOfString(List<string> values, BlittableJsonRead
                 {
                     switch (itr.Current)
                     {
-                        //todo maciej
-                        // case BlittableJsonReaderObject item:
-                        //     var clone = item.CloneOnTheSameContext();
-                        //     builder.WriteEmbeddedBlittableDocument(clone.BasePointer, clone.Size);
-                        //     break;
-
+                        case BlittableJsonReaderObject bjro:
+                            throw new InvalidQueryException($"{nameof(Corax)} does not support searching over complex objects. If you want to find matching documents please compare them by their properties.");
```

Comment:
This is the case that complex objects are inside nested array, right? So it's very similar or the same case like when we throw `ThrowIndexingComplexObjectNotSupported()`. Aren't we able to detect that earlier?

Reply:
I think we're already doing it here: https://github.com/ravendb/ravendb/pull/14044/files#diff-32e55cbc24468e2b2d3ad17fca814f4ac0922689c702c519c6a21c456f532185R312 

So my question is a bit different - does a user ever gets this error?

Reply:
The case when user tries to query on the not indexed field. I've introduced new exceptions due to that (throws when user is trying to query on not existing or not indexed field). I left this exception but moved it to the end in the `switch` tree (maybe there are some cases when we will hit that). Please check it.